Challenge cards

andrealbiac opened the pull request:
PR for issue #242 

The idea was something like this:
<img width="766" height="630" alt="Captura de pantalla 2025-08-04 a las 15 36 34" src="https://github.com/user-attachments/assets/968f5765-6d46-465a-aad1-75de27298b96" />

But I did this for now, with no background and two colors in the card
<img width="1298" height="701" alt="Captura de pantalla 2025-08-04 a las 15 36 19" src="https://github.com/user-attachments/assets/42dcf80f-cf57-4436-8932-3f41b0fa1435" />

---

I'm not sure the card component is properly made because I don't have access to the database (I was having some problems with Docker, might try again)- can you give me some feedback on the card component/data fetching please? @rin-st 

Still working on it:

- [x] want to tweak the dark mode
- [x] make sure the card component is correct
- [x] figure out why layer 2 link/page doesn't work
- [x] I'll add the builder ideas cards when the rest is OK

(+ Let me know if you're not a fan of commented section titles, it just helps me see through the code)

Thanks guys!

rin-st reviewed (commented):
> I'm not sure the card component is properly made because I don't have access to the database (I was having some problems with Docker, might try again)- can you give me some feedback on the card component/data fetching please? @rin-st 

Thanks Andrea, component and fetching are good!

>  figure out why layer 2 link/page doesn't work

it doesn't work because it's "Coming soon" challenge, we don't have a page for it. So, for it we need a way to disable card and maybe show something like a "Coming soon" badge. You can check if it's layer 2 challenge like by comparing `challenge.id` and  `ChallengeId.DEPLOY_TO_L2` 

Also, see other comments

rin-st commented on packages/nextjs/app/start/_components/ChallengeCard.tsx:
```diff
@@ -0,0 +1,37 @@
+  const firstSentence = challenge.description.split(".")[0] + ".";
+
+  return (
+    <a href={`https://speedrunethereum.com/challenge/${challenge.id}`} className="block">
```
> Updated this, now I can't open any challenge page at all but might be my problem?

For me working good, I think it's not working for you since you don't use docker

rin-st commented:
Thanks for the updates, Andrea!

> I think we should change this CTA section in this PR. Instead of Ch0, we should make the user register (connect wallet + register) and redirect them to their profile. There, they will see all the uncompleted challenges. You can take a look to the WelcomeBanner.tsx component for some idea for the copy + implementation.
> Let's keep tinkering!

> What do you think of the copy?

When user is connected we still showing this

<img width="1119" height="328" alt="image" src="https://github.com/user-attachments/assets/c4bdfc6e-18a0-4b0a-acae-0533dbfdaa26" />

Maybe it's better to show "Keep learning!" and "Portfolio" button? cc @carletex 

Also, need to fix loading state and redirect to profile after registering (or maybe just show portfolio button)

https://github.com/user-attachments/assets/3165808b-4ddc-492e-925c-5715e9e0c99b

carletex commented:
> Maybe it's better to show "Keep learning!" and "Portfolio" button? cc @carletex

Yep, or even the verbose version "Go to your Portfolio"

In any case, no worries about the copy, I'll do a pass before merging and might tweak some little things.

> redirect to profile after registering

I think redirecting is a nice a idea, now that we have the "gamified portfolio"

andrealbiac commented:
Added a new style for the 'Build idea' cards, and tweaked dark mode of both. Also changed layout to 2 columns on mobile cause it felt too long otherwise.

<img width="550" height="316" alt="Captura de pantalla 2025-08-07 a las 15 40 49" src="https://github.com/user-attachments/assets/9c646565-c4b3-403e-a25a-79b2c3a2e558" />
<img width="550" height="330" alt="Captura de pantalla 2025-08-07 a las 15 40 53" src="https://github.com/user-attachments/assets/4dd1b1f9-71f7-4621-9e4c-3cd63f924fdb" />

---

The only thing is now the 'build idea' and 'coming soon' logic is not dynamic, so if there's a new Build idea or the Coming soon challenge is ready and not coming soon anymore, we need to change the label manually
Maybe it's good for now and I/we can think of a more low-maintenance solution in the future 👍🏻 or maybe this is fine

andrealbiac commented on packages/nextjs/app/start/_components/ConnectAndRegisterSection.tsx:
```diff
@@ -0,0 +1,104 @@
+
+  // If still loading user data, show nothing
+  if (isLoadingUser) {
+    return null;
```
Updated to use the spinner. It doesnt show the button + spinner but I think it's fine. It also charges so quickly for me that I almost can't see it

andrealbiac commented on packages/nextjs/app/start/page.tsx:
```diff
@@ -161,29 +171,46 @@ const StartLandingPage = () => {
+              <SpaceshipIcon />
+              <h2 className="m-0 text-center text-2xl font-medium md:text-4xl">Ready?</h2>
+            </div>
+            <Image
```
Agree, I changed it for SRE with the same guys. I like the zebra and buffalo

<img width="638" height="540" alt="Captura de pantalla 2025-08-08 a las 12 10 57" src="https://github.com/user-attachments/assets/0faf34ec-b391-4e57-95d9-d5dac7f4c996" />

rin-st reviewed (approved):
Lgtm, thank you @andrealbiac !

andrealbiac commented:
> * [button hover](https://github.com/BuidlGuidl/SpeedRunEthereum-v2/pull/268/commits/014914e175032184f73a6294025555b1835d4364): Added a subtle hover

Looks good

> * [Build idea sorting](https://github.com/BuidlGuidl/SpeedRunEthereum-v2/pull/268/commits/049eeddd8f9c78d2c6a5fedfe377df4b4a7d538d): Sorting the builds ideas at the end (if not, stablecoins challenge was apprearing after the builds ideas)

Ahh weird! That didn't happen for me, even with the database



> * [isComingSoon and isBuildIdea](https://github.com/BuidlGuidl/SpeedRunEthereum-v2/pull/268/commits/38baaa722d97d6a292ca116c484d2b589ba6d5fe): Dyanmically detect isComingSoon and isBuildIdea

Amazing!

> * I think we need to come with a canonical solution (e.g. Always linking to Github), but this fix should work for now.

To me it makes more sense that we link everything to the internal page? And maybe we add a TG button in the page
So we're still inside SRE while reading more about the challenge.. not sure if the TG channel helps so early on, Just thinking out loud


Thanks for the tweaks! 🙌🏻

rin-st commented:
Thanks for updates @carletex !

> [button hover](https://github.com/BuidlGuidl/SpeedRunEthereum-v2/pull/268/commits/014914e175032184f73a6294025555b1835d4364): Added a subtle hover

Added hover to "Go your portfolio" button too

Lgtm!
